stan/math/prim/fun/singular_values.hpp

```diff
@@ -17,7 +17,8 @@ namespace math {
  * @param m Specified matrix.
  * @return Singular values of the matrix.
  */
-template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr>
+template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr,
+          require_not_vt_var<EigMat>* = nullptr>
```

Comment:
Alright so we'll do the varmat conversion here first. Along with `Eigen::Matrix<var, R, C>`, we just added a new autodiff type to Stan (this [one](https://github.com/stan-dev/design-docs/blob/master/designs/0005-static-matrices.md)). We don't have developer docs up so lemme walk you through how to convert this.

First change `require_not_vt_var<EigMat>` to `require_not_st_var<EigMat>`.

The first says "Require the value type of the template parameter to not be a var". Varmats are `var_value<Eigen::Matrix<T, R, C>>` types, and the value type of a `var_value<Eigen::Matrix<T, R, C>>` is `Eigen::Matrix<T, R, C>`. The scalar type though is `var`.

So for this to work with varmat we want to say "Require the scalar type of the template parameter to not be a var"